src/main/java/org/spongepowered/api/event/network/ClientConnectEvent.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.spongepowered.api.event.network;
+
+import org.spongepowered.api.GameProfile;
+import org.spongepowered.api.event.entity.player.PlayerJoinEvent;
+import org.spongepowered.api.text.Text;
+
+/**
+ * Fired when a client (player) attempts to connect to the server.
+ *
+ * <p>This event fires during the login process when the login state is
+ * {@code READY_TO_ACCEPT} - after the token has been verified but before the
+ * connection state switches to 'play'. See
+ * http://wiki.vg/Protocol#Login_Success for the protocol info.</p>
+ *
+ * <p>The server may have cancelled the event if the player is banned or
+ * whitelisted. Be sure to set {@code ignoreCancelled=false} in the
+ * {@code @Subscribe} annotation to receive the event in this case.</p>
+ *
+ * <p>Cancelling the event will prevent the player from joining and show
+ * {@link #getDisconnectMessage} to the player.</p>
+ *
+ * @see PlayerJoinEvent
+ */
+public interface ClientConnectEvent extends PlayerConnectionEvent, ConnectEvent {
+
+    /**
+     * Gets the disconnect message that will show to the client if the event is
+     * cancelled.
+     *
+     * @return The disconnect message
+     */
+    Text getDisconnectMessage();
```

Comment:
Unfortunately not, because a `CommandSource` is not available. `MessageEvent` is actually for the other way around.
> Describes events when a CommandSource sends a Text message.